### PR TITLE
 tidb master upgrade golang version to go1.18

### PR DESCRIPTION
**Pull request by purelind**
* tidb ci in master branch need upgrade go version to go1.18

### CODE REVIEW

**Comment by purelind** [comment]
hold until this issue closed  https://github.com/pingcap/tidb/issues/32874

File: jenkins/pipelines/ci/tidb/tidb_ghpr_check.groovy [hunk 1]
```diff
@@ -14,84 +14,83 @@ def notRun = 1
...
+                    containerTemplate(
+                        name: 'golang', alwaysPullImage: true,
+                        image: "${POD_GO_IMAGE}", ttyEnabled: true,
+                        resourceRequestCpu: '4000m', resourceRequestMemory: '8Gi',
+                        command: '/bin/sh -c', args: 'cat',
+                        envVars: [containerEnvVar(key: 'GOPATH', value: '/go')]     
+                    )
+            ],
+            volumes: [
+                    nfsVolume(mountPath: '/home/jenkins/agent/ci-cached-code-daily', serverAddress: '172.16.5.22',
```
Comment (dianqihanwangzi) [inline jenkins/pipelines/ci/tidb/tidb_ghpr_check.groovy:62]: this cache code never used

File: jenkins/pipelines/ci/tidb/tidb_ghpr_check_2.groovy [hunk 2]
```diff
@@ -108,7 +78,7 @@ def run_with_pod(Closure body) {
             containers: [
                     containerTemplate(
                             name: 'golang', alwaysPullImage: false,
-                            image: "${pod_go_docker_image}", ttyEnabled: true,
```
Comment (dianqihanwangzi) [inline jenkins/pipelines/ci/tidb/tidb_ghpr_check_2.groovy:111]: should change `go get gotest.tools/gotestsum` to `go install gotest.tools/gotestsum@latest`

File: jenkins/pipelines/ci/tidb/tidb_ghpr_unit_test.groovy [hunk 3]
```diff
@@ -17,55 +17,34 @@ if (params.containsKey("release_test")) {
...
-        println "targetBranch=${ghprbTargetBranch}  >= ${releaseBranchUseGo1160}"
-    }
+GO_VERSION = "go1.18"
+POD_GO_IMAGE = ""
+GO_IMAGE_MAP = [
+    "go1.13": "hub.pingcap.net/jenkins/centos7_golang-1.13:latest",
+    "go1.16": "hub.pingcap.net/jenkins/centos7_golang-1.16:latest",
+    "go1.18": "hub.pingcap.net/jenkins/centos7_golang-1.18:latest",
+]
+POD_LABEL_MAP = {
```
Comment (dianqihanwangzi) [inline jenkins/pipelines/ci/tidb/tidb_ghpr_unit_test.groovy:27]: should use `[]`

File: jenkins/pipelines/ci/tidb/tidb_ghpr_mysql_test.groovy [hunk 4]
```diff
@@ -119,50 +119,29 @@ if (m3) {
...
-    }
-    return false
+GO_VERSION = "go1.18"
+POD_GO_IMAGE = ""
+GO_IMAGE_MAP = [
+    "go1.13": "hub.pingcap.net/jenkins/centos7_golang-1.13:latest",
+    "go1.16": "hub.pingcap.net/jenkins/centos7_golang-1.16:latest",
+    "go1.18": "hub.pingcap.net/jenkins/centos7_golang-1.18:latest",
+]
+POD_LABEL_MAP = {
```
Comment (dianqihanwangzi) [inline jenkins/pipelines/ci/tidb/tidb_ghpr_mysql_test.groovy:129]: should use `[]`

**Comment by purelind** [comment]
replay test ci:
1. build  https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_build/detail/tidb_ghpr_build/52770/pipeline/
2. check_dev  https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check/detail/tidb_ghpr_check/54655/pipeline
3. check_dev2  https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/69325/pipeline

**Comment by purelind** [comment]
related to https://github.com/pingcap/tidb/pull/33053

**Pull request merged**